### ISSUE TITLE
Winget fails to update PowerShell v7.4.4.0 to newer releases and is not recognized when using the --id parameter

### ISSUE THREAD

**Issue by keanesc** [PowerShell/PowerShell#24499]
### Prerequisites

- [X] Write a descriptive title.
- [ ] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

1. Install PowerShell using Winget as described by the **learn.microsoft.com** documentation
`> winget install --id Microsoft.PowerShell --source winget`

2. Try the following commands
`> winget upgrade --id Microsoft.PowerShell --source winget`
`> winget upgrade --name PowerShell --version 7.4.6.0`

![image](https://github.com/user-attachments/assets/410b5801-53f9-4596-a582-5aa0324d7e56)

### Expected behavior

```console
It should update to the new version and show the new version in Winget
```

### Actual behavior

```console
> winget upgrade --id Microsoft.PowerShell --source winget
No installed package found matching input criteria.
> winget upgrade --name PowerShell --version 7.4.6.0
No version found matching: 7.4.6.0
```

### Error details

```console
PS> Get-Error

No output.
```

### Environment data

```powershell
❯ $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.4
PSEdition                      Core
GitCommitId                    7.4.4
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![image](https://github.com/user-attachments/assets/410b5801-53f9-4596-a582-5aa0324d7e56)

**Comment by MVPuknowme** [comment]
It looks like you're trying to upgrade PowerShell using the winget package manager. Let's walk through the process step-by-step to ensure you can successfully upgrade to the desired version:

### Steps to Upgrade PowerShell:

1. **Check Installed Version**: 
   - First, verify which version of PowerShell is currently installed. You can do this using:
     ```
     winget list --id Microsoft.PowerShell
     ```

2. **Search for Available Versions**:
   - Ensure that the desired version is available for upgrade:
     ```
     winget search PowerShell
     ```

3. **Upgrade PowerShell**:
   - If you want to upgrade to the latest version available, you can run:
     ```
     winget upgrade --id Microsoft.PowerShell
     ```
   - If a specific version is needed, make sure to specify it correctly:
     ```
     winget upgrade --id Microsoft.PowerShell --version <desired-version>
     ```

4. **Verify Installation**:
   - After the upgrade process, verify the installation:
     ```
     pwsh --version
     ```
   - This will confirm that the new version is successfully installed.

### Troubleshooting:
- **No Matching Package Found**: If you encounter errors about no matching packages, ensure that the package ID or version number is correct.
- **Administrative Privileges**: You might need to run your command prompt or terminal as an administrator to execute these commands successfully.

If you still face issues, please provide any specific error messages or behavior you're encountering

**Comment by keanesc** [comment]
Hi as mentioned above, any Winget command that uses the `--id` parameter returns the following error.

`No installed package found matching input criteria.`

And upon using the `name` parameter, it reports the following :

`no version found matching`

**Comment by mklement0** [comment]
On my Microsoft Windows 11 Pro (ARM 64-bit Processor; Version 22H2, OS Build: 22621.4317) machine with `winget --version` v1.8.1911 I do _not_ see your symptom.

(Note that your plain-text commands have an extraneous space between `--` and the option name, but seems to be just a posting artifact.)

Do you have the same problem with other packages too, or just the PowerShell one?

**Comment by alimbada** [comment]
I have the same issue
```
 Ammaar.Limbada  winget list --upgrade-available | sls PowerShell

PowerShell 7.4.5.0-x64                     Microsoft.PowerShell                  7.4.5.0         7.4.6.0         winget

 Ammaar.Limbada  winget update Microsoft.PowerShell
No installed package found matching input criteria.
 Ammaar.Limbada  winget upgrade Microsoft.PowerShell
No installed package found matching input criteria.
 Ammaar.Limbada  winget upgrade --id Microsoft.PowerShell
No installed package found matching input criteria.
```

It's specific to PowerShell. Other packages are updating fine.

**Comment by keanesc** [comment]
> Do you have the same problem with other packages too, or just the PowerShell one?

The issue only occurs with PowerShell.

P.S. After performing a clean install of Windows 11, the issue no longer occurs.

**Comment by alimbada** [comment]
> P.S. After performing a clean installation of Windows 11, the issue no longer occurs.

I'd rather not employ a brute force solution like that. Will you do that every time there's a problem upgrading the PowerShell package?

**Comment by keanesc** [comment]
Fair point, a clean install isn’t ideal. Reopening the issue to find a better solution.

**Comment by martinfrankjr** [comment]
I have the same issue.

Currently on `PowerShell v7.4.6.0`

It's specific to PowerShell. Other packages are updating fine.

From: `winget upgrade`
<img width="851" alt="Image" src="https://github.com/user-attachments/assets/1861ca1d-9383-4a17-bb50-56335752ed0a" />

From: `winget upgrade Microsoft.PowerShell`
![Image](https://github.com/user-attachments/assets/0598dd13-0eaf-4b95-9ef0-09a00676bdf9)

From: `winget list --id Microsoft.PowerShell`
![Image](https://github.com/user-attachments/assets/b1f49596-646b-438b-b8b1-e2afe5038955)

Let me know if I can provide any other details.

**Comment by martinfrankjr** [comment]
Found winget log:

[WinGet-2025-02-01-10-39-57.621.log](https://github.com/user-attachments/files/18629342/WinGet-2025-02-01-10-39-57.621.log)

**Comment by stuarthillary** [comment]
Does specifying the architecture help when trying to upgrade?

`winget upgrade --id Microsoft.Powershell --architecture x64`

The [manifest](https://github.com/microsoft/winget-pkgs/blob/master/manifests/m/Microsoft/PowerShell/7.5.0.0/Microsoft.PowerShell.installer.yaml) here lists all the available architectures.

**Comment by kilasuit** [comment]
FYI - This was something that I had seen in diagnosing other similar issues & [did a detailed writeup in this comment](https://github.com/PowerShell/PowerShell/issues/24237#issuecomment-2341816199)

TL:DR if you don't wanna read that comment - the fix is to run 
`winget install --id Microsoft.PowerShell -s winget -v 7.5.0.0 --uninstall-previous --force`
